crates/aggregator/src/lib.rs

```diff
@@ -78,7 +73,13 @@ impl Aggregator {
     /// # Returns
     ///
     /// * `Self` - The aggregator
-    pub async fn new(config: IncredibleConfig) -> Result<Self, AggregatorError> {
+    ///
+    /// # Errors
+    ///
+    /// * `AggregatorError` - The error that occurred
+    pub async fn new(
+        config: IncredibleConfig,
+    ) -> Result<(Self, AggregateReceiver), AggregatorError> {
```

Comment:
Why the return values change? `Self` should wrap the `AggregateReceiver` or provide a better abstraction than `Aggregator::new`.

Reply:
Let's leave this for another PR as discussed